nvbench/main.cuh

```diff
@@ -25,8 +25,16 @@
 #include <nvbench/option_parser.cuh>
 #include <nvbench/printer_base.cuh>
 
+#include <cstdlib>
 #include <iostream>
 
+static int _nvbench_env [[maybe_unused]] = []() -> int
```

Comment:
question: Initialization of global variables has undefined order. Since we control the main function, having the `setenv` right at the beginning of main (and before any of the CUDA calls) seems like more sound approach. Is there any reason to set the environment variable before main?

Reply:
> Is there any reason to set the environment variable before main?

I tried that first, but setting the env var at the start of main does not work -- it appears that CUDA is initialized prior to main. Using this static initialization works in my tests, but I am also concerned about initialization order and want to test it on a larger set of benchmarks to see if this is workable solution. Otherwise we may need to find a way to restart the CUDA context or something.

Reply:
Updated to remove the pre-main usage of CUDA (thanks @pciolkosz!) and moved the setenv into main -- which works now :)